File: packages/npm/src/steps/telemetry.ts
const TELEMETRY_URL =
  process.env.GOODVIBES_TELEMETRY_URL ?? 'https://goodvibes-telemetry.igiokas.workers.dev/'

const truthy = (v: string | undefined) => ['1', 'true', 'yes'].includes((v ?? '').trim().toLowerCase())

// CI services set CI to true, 1 or their own name; only an explicit 0 or false means not CI. The pip package uses the same rule.
const isCi = (v: string | undefined) => !['', '0', 'false'].includes((v ?? '').trim().toLowerCase())

export function telemetryOptedOut(env: NodeJS.ProcessEnv = process.env): boolean {
  return truthy(env.DO_NOT_TRACK) || truthy(env.GOODVIBES_NO_TELEMETRY) || isCi(env.CI)
}

export async function sendTelemetry(): Promise<void> {
  if (telemetryOptedOut()) return

  const { randomUUID } = await import('node:crypto')
  const id = randomUUID() // per-invocation, never stored to disk (TEL-02)
  const ac = new AbortController()
  const timer = setTimeout(() => ac.abort(), 1_000)
  timer.unref()   // won't prevent process exit if fetch somehow lingers
  try {
    await fetch(TELEMETRY_URL, {
      method: 'POST',
      body: null,
      signal: ac.signal,
      headers: { 'X-Request-Id': id },
    })
  } catch {
    // ponytail: silent on error — network failure must not affect init
  } finally {
    clearTimeout(timer)
  }
}

File: packages/pip/src/goodvibes_cli/steps/telemetry.py
"""Anonymous telemetry for goodvibes init — fire-and-forget install counter."""
import os
import threading
import urllib.request
import uuid as _uuid

TELEMETRY_URL = os.environ.get(
    "GOODVIBES_TELEMETRY_URL",
    "https://goodvibes-telemetry.igiokas.workers.dev/",
)


def opted_out() -> bool:
    return (
        any(os.environ.get(v, "").strip().lower() in ("1", "true", "yes") for v in ("DO_NOT_TRACK", "GOODVIBES_NO_TELEMETRY"))
        # CI services set CI to true, 1 or their own name; only an explicit 0 or false means not CI. npm uses the same rule.
        or os.environ.get("CI", "").strip().lower() not in ("", "0", "false")
    )


def _fire(request_id: str) -> None:
    req = urllib.request.Request(
        url=TELEMETRY_URL,
        method="POST",
        headers={"X-Request-Id": request_id},
    )
    try:
        urllib.request.urlopen(req, timeout=5)
    except OSError:
        pass  # ponytail: silent on network error — must not affect init


def start_telemetry_thread() -> threading.Thread | None:
    if opted_out():
        return None
    request_id = str(_uuid.uuid4())
    t = threading.Thread(target=_fire, args=(request_id,), daemon=True)
    t.start()
    return t
